cartography/intel/aws/iam.py

```diff
@@ -364,38 +385,198 @@ def load_user_access_keys(neo4j_session, user_access_keys, aws_update_tag):
+        if "NotAction" in stmt:
+            stmt["NotAction"] = ensure_list(stmt["NotAction"])
+        if "NotResource" in stmt:
+            stmt["NotResource"] = ensure_list(stmt["NotResource"])
+        if "Condition" in stmt:
+            stmt["Condition"] = json.dumps(ensure_list(stmt["Condition"]))
+    return statements
+
+
+@timeit
+def load_policy(neo4j_session, policy_id, policy_name, policy_type, principal_arn, aws_update_tag):
+    injest_policy = """
+    MERGE (policy:AWSPolicy{id: {PolicyId}})
+    ON CREATE SET
+    policy.firstseen = timestamp(),
+    policy.type = {PolicyType},
+    policy.name = {PolicyName}
+    SET
+    policy.lastupdated = {aws_update_tag}
+    WITH policy
+    MATCH (principal:AWSPrincipal{arn: {PrincipalArn}})
+    MERGE (policy) <-[r:POLICY]-(principal)
+    SET r.lastupdated = {aws_update_tag}
+    """
+    neo4j_session.run(
+        injest_policy,
+        PolicyId=policy_id,
+        PolicyName=policy_name,
+        PolicyType=policy_type,
+        PrincipalArn=principal_arn,
+        aws_update_tag=aws_update_tag,
+    ).consume()
+
+
+@timeit
+def load_policy_statements(neo4j_session, policy_id, policy_name, statements, aws_update_tag):
+    injest_policy_statement = """
+        MATCH (policy:AWSPolicy{id: {PolicyId}})
+        WITH policy
+        UNWIND {Statements} as statement_data
+        MERGE (statement:AWSPolicyStatement{id: statement_data.id})
+        SET
+        statement.effect = statement_data.Effect,
+        statement.action = statement_data.Action,
+        statement.notaction = statement_data.NotAction,
+        statement.resource = statement_data.Resource,
+        statement.notresource = statement_data.NotResource,
+        statement.condition = statement_data.Condition,
+        statement.sid = statement_data.Sid,
+        statement.lastupdated = {aws_update_tag}
+        MERGE (policy)-[r:STATEMENT]->(statement)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+        """
+    neo4j_session.run(
+        injest_policy_statement,
+        PolicyId=policy_id,
+        PolicyName=policy_name,
+        Statements=statements,
+        aws_update_tag=aws_update_tag,
+    ).consume()
+
+
+@timeit
+def load_policy_data(neo4j_session, policy_map, aws_update_tag):
+    for principal_arn, policy_list in policy_map.items():
+        logger.debug(f"Syncing IAM inline policies for principal {principal_arn}")
+        for policy in policy_list:
+            name = policy.name
+            policy_id = f"{principal_arn}/inline_policy/{name}"
+            try:
+                statements = _transform_policy_statements(
+                    policy.policy_document["Statement"], policy_id,
+                )
+                load_policy(neo4j_session, policy_id, name, "inline", principal_arn, aws_update_tag)
+                load_policy_statements(neo4j_session, policy_id, name, statements, aws_update_tag)
+            except ClientError:
```

Comment:
I want to get away from thinking of roles, groups and users seperatly and instead think of them as AWSPrincipals. So while the upstream requires separate methods when it get to loading it they are all just policies attached to AWSPrincipals. You are going to end up with identical functions. 
I think your concerns could be mitigated with some data checking in the load_policy function so it will fail fast on developer machines. However splitting them up means developers would make changes to role inline policies and not group inline policies which is not what I want.